backend-lib/src/main/rust/lib.rs

```diff
@@ -438,17 +421,128 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_createAcc
                 }
             })?;
 
-        let (account_id, usk) = db_data
-            .create_account(&seed, &birthday)
+        let account_name = java_string_to_rust(env, &account_name);
+        let key_source = java_nullable_string_to_rust(env, &key_source);
+
+        let (account_uuid, usk) = db_data
+            .create_account(
+                &account_name,
+                &seed,
+                &birthday,
+                key_source.as_ref().map(|s| s.as_ref()),
+            )
             .map_err(|e| anyhow!("Error while initializing accounts: {}", e))?;
 
-        let account = db_data.get_account(account_id)?.expect("just created");
-        let account_index = match account.source() {
-            AccountSource::Derived { account_index, .. } => account_index,
-            AccountSource::Imported { .. } => unreachable!("just created"),
-        };
+        Ok(encode_usk(env, account_uuid, usk)?.into_raw())
+    });
+    unwrap_exc_or(&mut env, res, ptr::null_mut())
+}
 
-        Ok(encode_usk(env, account_index, usk)?.into_raw())
+/// Tells the wallet to track an account using a unified full viewing key.
+///
+/// Returns details about the imported account, including the unique account identifier for
+/// the newly-created wallet database entry. Unlike the other account creation APIs
+/// ([`Self::create_account`] and [`Self::import_account_hd`]), no spending key is returned
+/// because the wallet has no information about how the UFVK was derived.
+///
+/// Certain optimizations are possible for accounts which will never be used to spend funds. If
+/// `spending_key_available` is `false`, the wallet may choose to optimize for this case, in
+/// which case any attempt to spend funds from the account will result in an error.
```

Comment:
It's unclear whether this should apply to the hardware wallet case. Does `spending_key_available` mean:
1. That the spending key is unavailable to the SDK (but the SDK might still have to track other things necessary to make proofs, for example), or
2. That the account will never be spent from, and so no spend-related information needs to be tracked?

From the description in this paragraph I would assume 2, but from the name `spending_key_available` I would assume 1.